HIRS_Utils/src/main/java/hirs/utils/Functional.java

```diff
@@ -38,4 +43,25 @@ private Functional() {
 
         return selectedItems;
     }
+
+    /**
+     * This method can be used to compare the distinguished names given from
+     * certificates.  This compare uses X500Name class in bouncy castle, which
+     * compares the RDNs and not the string itself.  The method will check
+     * for '+' and replace them, X500Name doesn't do this.
+     *
+     * @param nameValue1 first general name to be used
+     * @param nameValue2 second general name to be used
+     * @return true if the values match based on the RDNs, false if not
+     */
+    public static boolean x500NameCompare(final String nameValue1, final String nameValue2) {
```

Comment:
This should go into a new Utility class like `X500Utils` or `BouncyCastleUtils`. Based on the only other method in here this class looks to have been designed to support methods often natively supported by functional programming languages, such as select, take, map, reduce, etc.

Reply:
Additionally, whatever class this goes into should have a new test class written to test this comparison. We should have the test certificates to support a Unit Test of this.